contracts/src/BoundlessMarket.sol

```diff
@@ -224,78 +225,114 @@ contract BoundlessMarket is
         // NOTE: Signature checks and recursive verification happen inside the assessor.
         bytes32[] memory requestDigests = new bytes32[](1);
         requestDigests[0] = fill.requestDigest;
-        bytes32 assessorJournalDigest =
-            sha256(abi.encode(AssessorJournal({requestDigests: requestDigests, root: claimDigest, prover: prover})));
+        bytes32 assessorJournalDigest = sha256(
+            abi.encode(
+                AssessorJournal({
+                    requestDigests: requestDigests,
+                    selectors: assessorReceipt.selectors,
+                    root: claimDigest,
+                    prover: assessorReceipt.prover
+                })
+            )
+        );
+        // Verification that the provided seal matches the required selector.
+        if (assessorReceipt.selectors.length == 1 && assessorReceipt.selectors[0].value != bytes4(fill.seal[0:4])) {
```

Comment:
It's true that there would need to be a bug in the assessor, or a vulnerability somewhere in the proving system for `> 0` and `== 1` to be functionally different here, but I still think we should use the slightly more robust version of this check, and add a note.
```suggestion
        // NOTE: Assessor guest ensures that the number of selectors <= the number of request digests in the journal.
        if (assessorReceipt.selectors.length > 0 && assessorReceipt.selectors[0].value != bytes4(fill.seal[0:4])) {
```